lib/_tls_legacy.js

```diff
@@ -585,14 +585,12 @@ function onhandshakestart() {
   ssl.lastHandshakeTime = now;
   if (first) return;
 
-  if (++ssl.handshakes > tls.CLIENT_RENEG_LIMIT) {
+  if (++ssl.handshakes > tls.CLIENT_RENEG_LIMIT && self.cleartext) {
```

Comment:
I'm also -1 on this change unless we can verify why it was setImmediate and not nextTick before. @indutny any reason you chose to setImmediate here?

Also note that this method fired the event in the next tick anyway so in terms of consistency I think this is fine the way it is.